packages/flutter_tools/lib/src/ios/xcodeproj.dart

```diff
@@ -48,25 +49,91 @@ Future<void> updateGeneratedXcodeProperties({
   bool useMacOSConfig = false,
   bool setSymroot = true,
 }) async {
+  final List<String> xcodeBuildSettings = _xcodeBuildSettingsLines(
+    project: project,
+    buildInfo: buildInfo,
+    targetOverride: targetOverride,
+    useMacOSConfig: useMacOSConfig,
+    setSymroot: setSymroot
+  );
+
+  _updateGeneratedXcodePropertiesFile(
+    project: project,
+    xcodeBuildSettings: xcodeBuildSettings,
+    useMacOSConfig: useMacOSConfig,
+  );
+
+  _updateGeneratedModuleBuildPhaseScript(
+    project: project,
+    xcodeBuildSettings: xcodeBuildSettings,
+    useMacOSConfig: useMacOSConfig,
+  );
+}
+
+void _updateGeneratedXcodePropertiesFile({
+  @required FlutterProject project,
+  @required List<String> xcodeBuildSettings,
+  bool useMacOSConfig = false,
+}) {
   final StringBuffer localsBuffer = StringBuffer();
 
   localsBuffer.writeln('// This is a generated file; do not edit or check into version control.');
+  xcodeBuildSettings.forEach(localsBuffer.writeln);
+  final File generatedXcodePropertiesFile = useMacOSConfig
+    ? project.macos.generatedXcodePropertiesFile
+    : project.ios.generatedXcodePropertiesFile;
+
+  generatedXcodePropertiesFile.createSync(recursive: true);
+  generatedXcodePropertiesFile.writeAsStringSync(localsBuffer.toString());
+}
+
+void _updateGeneratedModuleBuildPhaseScript({
+  @required FlutterProject project,
+  @required List<String> xcodeBuildSettings,
+  bool useMacOSConfig = false,
+}) {
+  final StringBuffer localsBuffer = StringBuffer();
+
+  localsBuffer.writeln('#!/bin/sh');
+  localsBuffer.writeln('# This is a generated file; do not edit or check into version control.');
+  for (String line in xcodeBuildSettings) {
```

Comment:
Curious, when would a user need to modify these?  It's all FLUTTER_-type generated variables.

Reply:
One thing I could imagine is I was add-to-app xcode running etc, then I want to try using a local engine (which seems to work today). I think you can cause that to happen by running `flutter build ios --local_engine something` in the flutter module and then running the outer project again but it's not too clear how to trigger it from the outer project.

Reply:
I think should already be passed into flutter build because FLUTTER_ENGINE and LOCAL_ENGINE are being exported in this build script https://github.com/flutter/flutter/pull/36793/files#diff-36f7d4b99eeed90b423768f2148e3ab8R160, and then parsed in xcode_backend https://github.com/flutter/flutter/blob/master/packages/flutter_tools/bin/xcode_backend.sh#L131.

Reply:
I meant the stuff will indeed be passed into the re-entrant flutter call when it's in the script, but do the users (from the perspective of looking at the outer app project) find out how to change the script (since it's generated and says do not edit).

I just meant we can add a line of doc in this script that says if you want to modify the exported vars in this script, go into the inner flutter project and build it the way you want it to first, then your subsequent outer xcode project runs will follow those vars.

Reply:
They shouldn't alter the generated file, we need to be able to blow it away at any time and not care about migrations.  I'm changing the generated script to just export the environment variables.  They can change the environment variables in the build phase between that call and the xcode_backend call if they need a local engine for testing.

Reply:
I added a line in https://github.com/flutter/flutter/issues/32862 to make sure we document how to do this